src/NexusMods.App.UI/Pages/MyGames/MyGamesView.axaml.cs

```diff
@@ -12,21 +12,31 @@ public MyGamesView()
         InitializeComponent();
 
         this.WhenActivated(d =>
-        {
-            this.WhenAnyValue(view => view.ViewModel!.InstalledGames)
-                .BindToView(this, view => view.DetectedGamesItemsControl.ItemsSource)
-                .DisposeWith(d);
-            
-            this.WhenAnyValue(view => view.ViewModel!.SupportedGames)
-                .BindToView(this, view => view.SupportedGamesItemsControl.ItemsSource)
-                .DisposeWith(d);
-            
-            this.BindCommand(ViewModel, vm => vm.GiveFeedbackCommand, view => view.GiveFeedbackButton)
-                .DisposeWith(d);
-            
-            this.BindCommand(ViewModel, vm => vm.OpenRoadmapCommand, view => view.OpenRoadmapButton)
-                .DisposeWith(d);
-        });
+            {
+                this.WhenAnyValue(view => view.ViewModel!.InstalledGames)
+                    .BindToView(this, view => view.DetectedGamesItemsControl.ItemsSource)
+                    .DisposeWith(d);
+
+                this.WhenAnyValue(view => view.ViewModel!.SupportedGames)
+                    .BindToView(this, view => view.SupportedGamesItemsControl.ItemsSource)
+                    .DisposeWith(d);
+
+                this.BindCommand(ViewModel, vm => vm.GiveFeedbackCommand, view => view.GiveFeedbackButton)
+                    .DisposeWith(d);
+
+                this.BindCommand(ViewModel, vm => vm.OpenRoadmapCommand, view => view.OpenRoadmapButton)
+                    .DisposeWith(d);
+                
+                this.WhenAnyValue(view  => view.ViewModel!.InstalledGames)
+                    .Select(games => games.Count == 0)
+                    .Subscribe(isEmpty =>
+                        {
+                            NoGamesDetectedText.IsVisible = isEmpty;
+                            DetectedGamesItemsControl.IsVisible = !isEmpty;
+                        }
+                    )
+                    .DisposeWith(d);
```

Comment:
This only updates if the InstalledGames property is assigned, so if the entire collection is replaced, because the value of the property is the reference to the ObservableCollection.

Observe changes on the Count property of the observable collection instead. Since the collection is observable, the Count property should emit propertyChanged events.


```suggestion
                this.WhenAnyValue(view  => view.ViewModel!.InstalledGames.Count)
                    .Select(installedCount  => installedCount == 0)
                    .Subscribe(isEmpty =>
                        {
                            NoGamesDetectedText.IsVisible = isEmpty;
                            DetectedGamesItemsControl.IsVisible = !isEmpty;
                        }
                    )
                    .DisposeWith(d);
```

TBF, I don't think we actually detect if a game is getting installed while the app is open anyways, but worth making the code more correct.